driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -581,9 +583,9 @@ boolean setPool(HostConnectionPool pool) {
      * The connection should generally not be reused after that.
      */
     void release() {
-        HostConnectionPool pool = poolRef.get();
-        if (pool != null)
-            pool.returnConnection(this);
+        Owner owner = ownerRef.get();
+        if (owner instanceof HostConnectionPool)
```

Comment:
Found this in testing and think this may be an issue:  This only releases the Connection once, however even if the connection had many requests inFlight the host would believe it would have more requests inFlight then it actually does.

Consider the following case:
1. Create a Pool with core(1), max(2), maxSimultaneousRequests(100).
2. Acquire 101 connections on core, new connection 'extra1' will be created.  totalInFlight=101
3. Close core connection, it is released, totalInFlight = 100.  'extra1' becomes core.
4. Attempt to borrow a connection after reconnect interval.  Since totalInFlight=100 a new connection is spawned even though it shouldn't be.